pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -45,3 +45,12 @@ func (d *DualWriterMode1) List(ctx context.Context, options *metainternalversion
 
 	return legacy.List(ctx, options)
 }
+
+func (d *DualWriterMode1) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
```

Comment:
in general i really like the approach of having separate modes. it feels clean and easy to understand what should be happening, but for modes 1 & 4 i was wondering if there is a benefit to having a wrapper, or if the underlying storage should be used directly?

my question is mainly for mode 1, because adding a wrapper around the legacy storage will change the API discovery information that is exposed because it will look like delete is now implemented by the legacy storage. this is probably not a huge deal, but some k8s clients (velero for example) rely on API discovery information to drive behavior

Reply:
Yeah, in the other modes all methods will be supported so I think it's fine for mode 1 also. Honestly I think it will streamline things significantly if we require that all legacy storages implement every method, even if they return an error saying it's not actually supported.

Reply:
Makes sense. This should be a future improvement to work on. Tracking it here as a nice to have for now: https://github.com/grafana/search-and-storage-team/issues/17